rpm/files_test.go

```diff
@@ -0,0 +1,106 @@
+package rpm
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/quay/claircore"
+	"github.com/quay/claircore/test"
+	"github.com/quay/zlog"
+)
+
+var (
+	cache = map[string]map[string]struct{}{}
+	refs  = map[string]int{}
+
+	testFileCache = &filesCache{
+		c:    cache,
+		refs: refs,
+	}
+)
+
+var rpmFilesTestcases = []struct {
+	name     string
+	isRPM    bool
+	filePath string
+}{
+	{
+		name:     "Found Node",
+		isRPM:    true,
+		filePath: "usr/lib/node_modules/npm/node_modules/safe-buffer/package.json",
+	},
+	{
+		name:     "Found Python",
+		isRPM:    true,
+		filePath: "usr/lib64/python3.9/site-packages/libcomps-0.1.18-py3.9.egg-info/PKG-INFO",
+	},
+	{
+		name:     "Not found",
+		isRPM:    false,
+		filePath: "usr/lib/node_modules/npm/node_modules/safe-buffer/package.jsonx",
+	},
+}
+
+func TestIsRPMFile(t *testing.T) {
+	ctx := zlog.Test(context.Background(), t)
+	a := test.NewCachedArena(t)
+	fc = testFileCache
+	layer := test.LayerRef{
+		Registry: "registry.access.redhat.com",
+		Name:     "ubi9/nodejs-18",
+		Digest:   `sha256:1ae06b64755052cef4c32979aded82a18f664c66fa7b50a6d2924afac2849c6e`,
+	}
+	a.LoadLayerFromRegistry(ctx, t, layer)
+	r := a.Realizer(ctx).(*test.CachedRealizer)
+	t.Cleanup(func() {
+		if err := r.Close(); err != nil {
+			t.Error(err)
+		}
+	})
+	t.Cleanup(func() {
+		fc.wg.Wait() // Wait for all goroutines to finish
+		err := checkCleanup(fc)
+		if err != nil {
+			t.Error(err)
+		}
+	})
+
+	realizedLayers, err := r.RealizeDescriptions(ctx, []claircore.LayerDescription{
+		{
+			Digest:    layer.Digest,
+			URI:       "http://example.com",
+			MediaType: test.MediaType,
+			Headers:   make(map[string][]string),
+		},
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+	for _, tt := range rpmFilesTestcases {
+		t.Run(tt.name, func(t *testing.T) {
+			ctx, cancel := context.WithCancel(zlog.Test(context.Background(), t))
+			t.Parallel()
+			isRPM, err := FileInstalledByRPM(ctx, &realizedLayers[0], tt.filePath)
+			if err != nil {
+				t.Fatal(err)
+			}
+			if tt.isRPM != isRPM {
+				t.Errorf("expected isRPM: %t, got isRPM: %t", tt.isRPM, isRPM)
+			}
+			cancel()
+		})
+	}
```

Comment:
instead of adding a `sync.WaitGroup` to `filesCache` can we just do this instead?

```suggestion
	var wg sync.WaitGroup
	for _, tt := range rpmFilesTestcases {
		wg.Add(1)
		t.Run(tt.name, func(t *testing.T) {
			defer wg.Done()
			ctx, cancel := context.WithCancel(zlog.Test(context.Background(), t))
			t.Parallel()
			isRPM, err := FileInstalledByRPM(ctx, &realizedLayers[0], tt.filePath)
			if err != nil {
				t.Fatal(err)
			}
			if tt.isRPM != isRPM {
				t.Errorf("expected isRPM: %t, got isRPM: %t", tt.isRPM, isRPM)
			}
			cancel()
		})
	}
```

If we do end up requiring it for testing purposes, can you add a comment in the struct to make it clear it's only there for testing purposes?

Reply:
ah ok just saw your other comment. We need to know when `gc` is called. Makes sense. Can you add a comment to the filesCache struct to make it clear it's just there for testing purposes? I was confused what it was doing there until I saw this. Thanks